src/Common/src/Interop/Unix/System.Native/Interop.RealPath.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        /// <summary>
+        /// Takes a path containing relative subpaths or links and returns the absolute path.
+        /// </summary>
+        /// <param name="path">The input path containing relative subpaths or links</param>
+        /// <param name="resultPath">The output path; will be null on error</param>
+        /// <returns>Returns 0 on succes and -1 on failure; on failure, resultPath will be null</returns>
+        internal static int RealPath(string path, out string resultPath)
```

Comment:
Should we just make it:
```C#
internal static string RealPath(string path);
```
?

non-null on success, null on failure, as is the case with the out arg as you have it.  Just seems like at that point the int return value is unnecessary.